packages/erxes-ui/src/components/richTextEditor/utils/getParsedMentions.tsx

```diff
@@ -1,10 +1,25 @@
 export const getParsedMentions = (data: any) => {
-  const mentions = (data.content || []).flatMap(getParsedMentions);
-  if (data.type === 'mention') {
-    mentions.push(data.attrs.id);
-  }
+  const mentionIds = new Set();
 
-  const uniqueMentions = [...new Set(mentions)] as any;
+  // Recursive function to traverse nested content
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **issue (complexity):** Consider simplifying the conditionals and overall approach to reduce complexity.

While the updated implementation successfully introduces recursion to handle nested content, it does introduce a higher level of complexity compared to the original approach. Specifically, the use of recursion and the handling of various conditions increase the cognitive load for readers and maintainers of the code. Additionally, the code now makes more specific assumptions about the data structure, which could affect its flexibility and adaptability to changes in the data model.

To maintain readability and ease of understanding, consider simplifying the conditionals and the overall approach. For instance, leveraging optional chaining and array methods like `.some()` can help streamline the checks for specific conditions, such as the presence of a mention mark. This not only reduces the number of lines but also makes the intent clearer at a glance.

Here's a suggested revision that aims to balance the need for traversing nested structures with the goal of keeping the code concise and readable:

```javascript
export const getParsedMentions = (data: any) => {
  const mentionIds = new Set();

  const traverseContent = (content: any[]) => {
    content.forEach((item: any) => {
      if (item.type === 'text' && item.marks?.some(mark => mark.type === 'spanMark' && mark.attrs?.['data-type'] === 'mention')) {
        item.marks.forEach((mark: any) => {
          if (mark.attrs?.['data-id']) {
            mentionIds.add(mark.attrs['data-id']);
          }
        });
      } else if (item.content) {
        traverseContent(item.content);
      }
    });
  };

  traverseContent(data.content || []);
  return Array.from(mentionIds);
};
```

This version aims to reduce complexity by making the conditions more concise and easier to understand, which could help in maintaining the code in the long run.